### ISSUE TITLE
suggestion to expand the android code samples for kotlin and static linking of vulkan

### ISSUE THREAD

**Issue by MagicPoulp** [KhronosGroup/Vulkan-Samples#219]
The android code sample uses only a native activity in C++.

You could improve the sample by adding the possibility to use a SurfaceView from kotlin, and only use a part of the screen.
I made a working code sample of the idea.
https://github.com/MagicPoulp/display_vulkan_from_kotlin_or_swift.git

Moreover, linking Vulkan dynamically is quite buggy (lots of compilation errors). It can be interesting to have a simpler way to link statically, with a flag.

The link below illustrates that lots of hacks might be needded to link dynamically vulkan, with missing symbols, and so on.
https://marcelbraghetto.github.io/a-simple-triangle/2019/06/16/part-17/

Using Vulkan allows to compute more efficiently and save battery life. So Vulkan on mobile is key.

**Comment by SaschaWillems** [comment]
> Moreover, linking Vulkan dynamically is quite buggy (lots of compilation errors). It can be interesting to have a simpler way to link statically, with a flag.

Can you elaborate on this? We're already using Volk as a function loader, and linking Vulkan static could introduce other headaches like loader incompatibilities.

From my experience I have heard little of problems with dynamic linking on Android, and also never ran into such problems myself. Or is that specific to using Vulkan with Kotlin?

**Comment by MagicPoulp** [comment]
Thanks for your interest. I remember your name with the first Vulkan samples when vulkan came out.

It was many months ago I wrote this so my memory is not fresh.

In the link above (marcel soemthing) there are many many failures and ways to fix them. IT shows the sort of mess, and thigns that depend on development software only.

I was using the latest LunarG code sample, the first one found on the internet, and there was no Volk at that time. Not that I remember. I had to use static, instead of dynamic to move on on the specific thing I was doing.

I was building using the Android NDK, they have their own C++ compiler. You can see exactly what I did if you clone my repo. It is fully documented, and runnable. The Android NDK has like a cmake build integrated. If you change my repo and copy paste exactly teh original sample with the dyanmic build, you will see the errors. In my code, it seems to my I just commented out.

The technology to make graphics programming directly available in apps is there (cf the surface view). In the Javascript world there is pIXI, you can drow with a turtle pointer and it graphics accelerate on its own using OpenGL. it would be could to have in kotlin a library where you can drow on the rectangle and it uses Vulkan.

It seems to me it that gaphics could then use less battery. And also I had a problem with lag inside the graph library charts_flutter
It needs to show thousands of points in a graph that is dynamic based on touch. Just a pet project. I am still hesitating if Vullkan would help or if I need to rewrite the flutter library to be smarter by sampling.
https://github.com/MagicPoulp/audio_analyzer/blob/master/lib/FftPlot.dart

If you make graphics programming more mainstream you have potentially million of dollars of consulting contracts for Lunar G and all the gurus of this cool open source stuff. The most straightforward first need lies in the company logo with an animation when the app starts.

**Comment by MagicPoulp** [comment]
I used the wrong repo. And in the wrong repo from LunarG, they do not use Volk. Therefore I will close this issue.
I used this one:
https://github.com/LunarG/VulkanSamples.git
Instead of yours that is this one.
https://github.com/KhronosGroup/Vulkan-Samples.git

In the LunarG version they use native activites only.
In the Khronos version, NFI binding from Java is used. ANd you use Volk.

UPDATE:
I was able to use Volk with only simple code change.
And I will tell it to the LunarG team.
https://github.com/MagicPoulp/display_vulkan_from_kotlin_or_swift.git
